public/app/features/dashboard-scene/v2schema/transformers.test.ts

```diff
@@ -0,0 +1,91 @@
+import { config } from '@grafana/runtime';
+import { CustomVariable, GroupByVariable } from '@grafana/scenes';
+import { DashboardV2Spec } from '@grafana/schema/dist/esm/schema/dashboard/v2alpha0/dashboard.gen';
+import { handyTestingSchema } from '@grafana/schema/dist/esm/schema/dashboard/v2alpha0/examples';
+import { DashboardWithAccessInfo } from 'app/features/dashboard/api/dashboard_api';
+
+import { transformSaveModelSchemaV2ToScene } from '../serialization/transformSaveModelSchemaV2ToScene';
+import { transformSceneToSaveModelSchemaV2 } from '../serialization/transformSceneToSaveModelSchemaV2';
+
+const defaultDashboard: DashboardWithAccessInfo<DashboardV2Spec> = {
+  kind: 'DashboardWithAccessInfo',
+  metadata: {
+    name: 'dashboard-uid',
+    namespace: 'default',
+    labels: {},
+    resourceVersion: '',
+    creationTimestamp: '',
+  },
+  spec: handyTestingSchema,
+  access: {},
+  apiVersion: 'v2',
+};
+
+jest.mock('@grafana/runtime', () => ({
+  ...jest.requireActual('@grafana/runtime'),
+  getDataSourceSrv: () => ({
+    getInstanceSettings: jest.fn(),
+  }),
+  config: {
+    ...jest.requireActual('@grafana/runtime').config,
+    bootData: {
+      settings: {
+        defaultDatasource: 'prometheus',
+        datasources: {
+          prometheus: {
+            name: 'datasource1',
+            meta: { id: 'prometheus' },
+            type: 'prometheus',
+          },
+        },
+      },
+    },
+  },
+}));
+
+describe('V2 Transformers', () => {
+  beforeAll(() => {
+    config.featureToggles.groupByVariable = true;
+  });
+
+  afterAll(() => {
+    config.featureToggles.groupByVariable = false;
+  });
+  it('should match original V2 Schema when transforming to scene and back to V2 Schema', async () => {
+    const dashV2Scene = transformSaveModelSchemaV2ToScene(defaultDashboard);
+    // Find and manually set options for CustomVariable
+    // Options are set based on query field using getValueOptions
+    const customVariable = dashV2Scene.state.$variables?.state.variables.find(
```

Comment:
Had to do this for custom and groupby variables because they rely on the getValueOptions that use Observable and the options array was always empty even though I was expecting a value